implementations/go/services/postbackService.go

```diff
@@ -2,26 +2,118 @@ package services
 
 import (
 	"encoding/json"
+	"fmt"
 	"soccergist/implementations/go/dataobject"
 	"soccergist/implementations/go/utility"
+	"strconv"
 )
 
 //HandlePostBackRecieved - function to handle the postbacks
 func HandlePostBackRecieved(postback dataobject.PostBack, sender dataobject.Sender) (response string) {
 	title := postback.Title
-	recipient := dataobject.Recipient{
+	payload := postback.Payload
+	var recipient dataobject.Recipient
+	var message dataobject.ResponseMessage
+	recipient = dataobject.Recipient{
 		ID: sender.ID,
 	}
+
+	fmt.Println(payload)
+
+	if payload == "league-table-postback" {
+		leagueID := 445
+		return loadLeagueTable(leagueID, sender)
+
+	} else {
+		message = dataobject.ResponseMessage{
+			Text: title + " coming soon",
+		}
+
+	}
+
+	jsonResponse := dataobject.JSONResponse{
+		Recipient: recipient,
+		Message:   message,
+	}
+	b, err := json.Marshal(jsonResponse)
+	utility.FailOnError(err, "Cannot Marshall response to json")
+	response = string(b)
+	return
+}
+
+func loadLeagueTable(leagueID int, sender dataobject.Sender) (response string) {
+	url := "http://api.football-data.org/v1/competitions/" + strconv.Itoa(leagueID) + "/leagueTable"
+	dataResponse := utility.SendGetRequest(url)
+	var decodedResponse map[string]interface{}
```

Comment:
What do you think of decoding directly into a struct with the fields you are interested in? That way you wouldn't need to have the interface typecasting you have below. You could use something like say
```go
type leagueTable struct {
  Standing      []struct {		
		Position       int    `json:"position"`
		TeamName       string `json:"teamName"`
		PlayedGames    int    `json:"playedGames"`
		Points         int    `json:"points"`
	} `json:"standing"`
}

decodedResponse := leagueTable{}
err := json.Unmarshal([]byte(dataResponse), &decodedResponse)
...
```

Reply:
hmmmm... yh i will try that out... i just wanted to avoid declaring structs for the entire data. So i decided to try out the map approach... did something similar in the java implementation...

Reply:
Yea, I see what you mean. Give this a try if it doesn't work then we can revert.

Reply:
alright... on it... thanks man